packages/app/src/cli/services/versions-list.ts

```diff
@@ -82,31 +82,26 @@ async function fetchAppVersions(
 }
 
 interface VersionListOptions {
-  app: AppInterface
-  apiKey?: string
-  reset: false
+  app: AppLinkedInterface
+  remoteApp: OrganizationApp
+  developerPlatformClient: DeveloperPlatformClient
   json: boolean
-  developerPlatformClient?: DeveloperPlatformClient
 }
 
 export default async function versionList(options: VersionListOptions) {
-  const result = await ensureVersionsListContext(options)
-  const developerPlatformClient = options.developerPlatformClient ?? result.developerPlatformClient
+  const {remoteApp, developerPlatformClient} = options
 
-  const {organizationId, title} = result.remoteApp
-
-  const {appVersions, totalResults} = await fetchAppVersions(developerPlatformClient, result.remoteApp, options.json)
-
-  const {businessName: org} = await fetchOrgFromId(organizationId, developerPlatformClient)
+  const {appVersions, totalResults} = await fetchAppVersions(developerPlatformClient, remoteApp, options.json)
 
+  const {businessName: org} = await fetchOrgFromId(remoteApp.organizationId, developerPlatformClient)
   if (options.json) {
     return outputInfo(JSON.stringify(appVersions, null, 2))
   }
 
   renderCurrentlyUsedConfigInfo({
     org,
-    appName: title,
-    configFile: isCurrentAppSchema(options.app.configuration) ? basename(options.app.configuration.path) : undefined,
+    appName: remoteApp.title,
+    configFile: basename(options.app.configuration.path),
```

Comment:
Why this change? What happens when the config file is old?

Reply:
it can't be old anymore, `linkedAppContext` forces you to re-link old config files so commands receive an updated schema always. (`AppLinkedInterface`)

In any case, the `path` is available in both the old and new schema.